src/TruncateExpandable.js

```diff
@@ -0,0 +1,10 @@
+import styled from 'styled-components'
+import Truncate from './Truncate'
+
+const TruncateExpandable = styled(Truncate)`
+  &:hover {
+    max-width: 10000px !important;
+  }
+`
+
+export default TruncateExpandable
```

Comment:
I debated having `expandable` be just a `variant` on Truncate, but I thought there might be occasions where you'd want to use `target` and `expandable` variants at the same time. Opting to extend the component seems to be the pattern you're following.

Reply:
[Similarly](https://github.com/primer/components/pull/685/files#r375403533), I feel like an `expandable` boolean prop on `Truncate` would be more ergonomic here. Neither really feels like a "variant." But I feel less strongly about this one than I do the `inline` one.

Reply:
Thanks for the feedback!

I opted for `variant` based on Brent Jackson's recommendation to [opt for variant enum props over boolean props](https://jxnblk.com/blog/defining-component-apis-in-react/). But agreed that it doesn't feel like a `variant` like `large` or `small`.

That said, the styles for these variations don't currently conflict with one another, so maybe they don't need to be separate components (or variants at all).